### ISSUE TITLE
MessageBus simple polling clobbers the session cookie

### ISSUE THREAD

**Issue by yemartin** [discourse/message_bus#257]
I have traced a weird bug to what I believe is an unwanted behavior in MessageBus. What seems to happen is this:

1. The browser sends an AJAX request.
2. Server side, the controller handling the AJAX request sets new session values, and sends some MessageBus messages.
3. Before the AJAX request completes, MessageBus (browser side) starts a new polling request. This request contains the current (soon to become **OLD**) session cookie.
3. Now the AJAX request completes, setting the session cookie to a **NEW** value.
4. Finally, the MessageBus polling request completes, but the response contains the now **OLD** session cookie, hence clobbering the values set by the AJAX request.

Note that this does not seem to happen with long-polling, only when simple polling, as long polling did not include a "Set-Cookie" header. We can see on this screenshot that simple polling requests do contain a "Set-Cookies" header:

<img width="429" alt="MessageBus_session_clobbering" src="https://user-images.githubusercontent.com/139002/144528114-7e087a74-2a46-48d3-b538-04ef370e59bf.png">

I have not seen this reported anywhere, so maybe I am missing something or have things set up incorrectly. Anyway, MessageBus polling should not have side effects other than handling messages. In particular, it should not be sending session cookies, as this can clobber session state set by other means, like in the above scenario.

I have a tentative fix here: https://github.com/mediweb/message_bus/commit/f3db0f24d671cbd272e8ed836daad4267f7145cf

With this change in place, here is the result with the same scenario as above: simple polling responses do not have a "Set-Cookie" header anymore:

<img width="429" alt="MessageBus_session_clobbering_fixed" src="https://user-images.githubusercontent.com/139002/144528565-0f84c646-c39f-4a66-8d93-098dfdd254db.png">

Should I open a pull request, or am I missing something?

**Comment by davidtaylorhq** [comment]
Hi @yemartin - thanks for the detailed report. What you're describing does sound like an issue. At Discourse, we almost exclusively use MessageBus in long-polling mode, so that's probably why this problem hasn't been noticed before.

If you're happy to open a PR with the fix, that would be great!

**Comment by yemartin** [comment]
Thank you for the confirmation @davidtaylorhq, I opened PR #258 

Changes I made since the initial tentative fix:
- replaced a hardcoded string with the corresponding Rack constant
- reduced code comment verbosity
- added a spec
- added a CHANGELOG entry

Note about the spec: unfortunately I was unable to test the behavior (the presence of a "Set-Cookie" header), as I was not able to make the Rack session mechanism set a session cookie in specs. Extra setup is probably needed, but I am out of my depth here. So I fell back on testing the correct Rack session option being set in the request.
